neo4j-ogm-docs/src/main/asciidoc/reference/nativetypes.adoc

```diff
@@ -0,0 +1,183 @@
+    <version>{ogm-version}</version>
+    <scope>runtime</scope>
+</dependency>
+
+<dependency>
+    <groupId>org.neo4j</groupId>
+    <artifactId>neo4j-ogm-bolt-native-types</artifactId>
+    <version>{ogm-version}</version>
+    <scope>runtime</scope>
+</dependency>
+----
+
+The setup for the embedded driver looks similar:
+
+.Support for native types temporal and spatial types with the embedded driver
+[source, xml, subs="attributes, specialcharacters"]
+----
+<dependency>
+    <groupId>org.neo4j</groupId>
+    <artifactId>neo4j-ogm-embedded-driver</artifactId>
+    <version>{ogm-version}</version>
+    <scope>runtime</scope>
+</dependency>
+
+<dependency>
+    <groupId>org.neo4j</groupId>
+    <artifactId>neo4j-ogm-embedded-native-types</artifactId>
+    <version>{ogm-version}</version>
+    <scope>runtime</scope>
+</dependency>
+
+<dependency>
+    <groupId>org.neo4j</groupId>
+    <artifactId>neo4j</artifactId>
+    <version>{neo4j-version}</version>
+    <scope>runtime</scope>
+</dependency>
+----
+
+The dependencies for a Gradle build are `'org.neo4j:neo4j-ogm-bolt-driver:{ogm-version}'` and  `'org.neo4j:neo4j-ogm-embedded-native-types:{ogm-version}'` respectively.
+
+[[reference:native-property-types:optin]]
+== Opt-in to use native types
+
+Using native types for temporal and spatial property types is a behaviour changing feature, as the default type conversion is turned off in that case and dates aren't neither written nor read to and from strings.
+Therefore it is an Opt-in feature.
+
+To opt-in, please first add the corresponding module for your <<reference:native-property-types:supported-drivers, driver>> and
+than use the new configuration property `use-native-types`:
+
+.Enabling the usage of native types
+[%autowidth.spread,frame="topbot",options="header"]
+|======================
+|ogm.properties   | Java Configuration
+
+a|
+[source, properties]
+----
+URI=bolt://neo4j:password@localhost
+use-native-types=true
+----
+
+a|
+[source, java]
+----
+Configuration configuration = new Configuration.Builder()
+        .uri("bolt://neo4j:password@localhost")
+        .useNativeTypes()
+        .build()
+----
+|======================
+
+The same works with a file based URI for the embedded driver.
+
+Once enabled, native types are used for all attributes of all node- and relationship-entities and also for all parameters passed through the OGM `Session` interface.
+
+[[reference:native-property-types:mapping]]
+== Mapping of native types
+
+The following table describes how Neo4j temporal and spatial property types are mapped to attributes of Neo4j-OGM entities:
+
+======
+[options="header", cols="m,"]
+|===
+| Neo4j type        | Neo4j-OGM type
+| Date | `LocalDate`
+| Time | `OffsetTime`
+| LocalTime | `LocalTime`
+| DateTime | `ZonedDateTime`
+| LocalDateTime | `LocalDateTime`
+| Duration | `TemporalAmount`*
+| Point | One of four different Neo4j-OGM  in `org.neo4j.ogm.types.spatial`**
```

Comment:
Is it wise to create four types? In future the CRS supported by Neo4j could number in the thousands. Do you want to have thousands of OGM types? Perhaps it works when we consider that all thousands probably fall into four groups defined by: geographic vs. cartesian and 2D vs 3D. In which case I guess you OGM types are: geographic-2d, geographic-3d, cartesian-2d and cartesian-3d?

Reply:
See general comment below. We deal exactly in those categories. Geographic variants only offer access to longitude and latitude while cartesian variants offer x and y. All variants offer their SRID.